interview_cake/balanced_binary_tree/sean/check_balanced.py

```diff
@@ -0,0 +1,26 @@
+class BinaryTree:
+    def __init__(self, left=None, right=None):
+        self.left = left
+        self.right = right
+        self.leaf = not bool(self.left or self.right)
+
+def check_balanced(tree, depth=0):
+    if not isinstance(tree, BinaryTree):
+        raise TypeError("you need a tree mang")
```

Comment:
When you return a type error, I recommend stating the expected type, the actual type, and the actual data.
At first this can seem somewhat redundant because the caller should know what they are calling with, but that often is not the case.

For example:
```
tree_list = get_trees()
for tree in tree_list:
    if not check_balanced(tree):
        balance(tree)
```
If there is a bug in get_trees() such that it returns something other than trees, it would be helpful to know what it is returning